packages/nearby-pois/src/reducer.ts

```diff
@@ -92,10 +92,16 @@ export default function nearbyPoisReducer(
 function deduplicateAndMergePoiList(
   prevPoiList: ListingPoi[],
   nextPoiList: ListingPoi[],
-) {
+): ListingPoi[] {
   const mergedPoiList = [...prevPoiList, ...nextPoiList]
-  return mergedPoiList.filter(
-    (poiInFilter, index) =>
-      index === mergedPoiList.findIndex((poi) => poi.id === poiInFilter.id),
-  )
+
+  const poiListMap = new Map<string, ListingPoi>()
+
+  for (const poi of mergedPoiList) {
+    if (!poiListMap.has(poi.id)) {
+      poiListMap.set(poi.id, poi)
+    }
+  }
+
+  return Array.from(poiListMap.values())
```

Comment:
요런식으로 선언형으로 바꿔주면 어떨까요?

```suggestion
  const poiMap = new Map([
    ...prevPoiList.map(poi => [poi.id, poi]),
    ...nextPoiList.map(poi => [poi.id, poi])
  ]);
 
  return Array.from(poiMap.values());;
```

Reply:
의견 주신 대로 수정했습니다! 0919db51867cbde503df5a36b593f5d614924144 여기서 타입도 추가했어요.